### DISCUSSION TITLE
Selve HA  intermediate position

### DISCUSSION THREAD

**Discussion by Edem123456** [Kannix2005/homeassistant-selve#21] [Q&A]
Hello,
Thank you very much for your work on roller blind control.
I am also currently using your project to control the blinds by using a USB-RF Gateway (iveo) and I have a question about controlling the blinds in intermediate positions. In HA I can only control it as open or close. How to use indirect control - partial opening? I was previously using the project _https://github.com/gpulido/python-selve_ and there were two intermediate items working. Is this also possible in your project? Maybe I'm doing something wrong and I don't see this possibility. Thank you in advance for your help.
Regards, Edward

**Comment by frankfrommelt** [comment]
Hi Edward,
well, I am using this integration with HA for some time with 6 Commeo blinds.
I have the option to open, close, set a percentage for the position and use two different tilt positions for the blinds.
So it might be this is not implemented for Iveo devices, as this seems to be the only difference.
I hope Stefan can give some insight here.
Take care
   Frank


Gesendet von Outlook für Android<https://aka.ms/AAb9ysg>
________________________________
From: Edem123456 ***@***.***>
Sent: Sunday, November 5, 2023 1:10:33 PM
To: Kannix2005/homeassistant-selve ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [Kannix2005/homeassistant-selve] Selve HA intermediate position (Discussion #21)


Hello,
Thank you very much for your work on roller blind control.
I am also currently using your project to control the blinds by using a USB-RF Gateway (iveo) and I have a question about controlling the blinds in intermediate positions. In HA I can only control it as open or close. How to use indirect control - partial opening? I was previously using the project https://github.com/gpulido/python-selve and there were two intermediate items working. Is this also possible in your project? Maybe I'm doing something wrong and I don't see this possibility. Thank you in advance for your help.
Regards, Edward

**Reply by Edem123456** [reply]
Hello Frank
my blinds are the Iveo system.  After reintegration in HA, I see options like Set tilt position, Open tilt, close tilt etc. - but these options do not work for me. I can only open or close the roller blind - no intermediate positions. 
In the previous integration, the Open tilt and close tilt positions triggered intermediate blind position settings. 
When I look at the source code, I miss entries that worked for Iveo, such as: 

    def moveIntermediatePosition1(self, automatic = False):
        self.executeCommand(CommandType.POSITION_1, automatic)

    def moveIntermediatePosition2(self, automatic = False):
        self.executeCommand(CommandType.POSITION_2, automatic)

Unfortunately, I don't know how to complete it or if it's even possible. 
The previous integration where it worked for Iveo - it blocks me from updating HA Core (2023.6.3 -> 2023.11.1), so I need to change the integration, but I miss the indirect setting of the roller blinds position.

It would be great if Stefan would be able to help in this case. Thank you in advance for your help.
Regards, Edward
